test/services/onlinetesting/OnlineTestServiceSpec.scala

```diff
@@ -82,6 +83,42 @@ class OnlineTestServiceSpec extends UnitSpec {
     }
   }
 
+  "processNextTestForSdipFsNotification" should {
+    "return a successful unit future if no test to notify is found" in new OnlineTest {
+      when(appRepositoryMock.findTestForSdipFsNotification(FailedSdipFsTestType)).thenReturn(Future.successful(None))
+
+      val result = underTest.processNextTestForSdipFsNotification(FailedSdipFsTestType).futureValue
+      result mustBe unit
+
+      verify(appRepositoryMock).findTestForSdipFsNotification(FailedSdipFsTestType)
+      verifyNoMoreInteractions(appRepositoryMock)
+      verifyZeroInteractions(cdRepositoryMock, emailClientMock)
+    }
+
+    "return a successful unit future if a test to notify is found" in new OnlineTest {
```

Comment:
this test name doesn't reflect everything the test is doing - we're finding the application to notify and then processing the notification, not just finding the test.